client/my-sites/stats/activity-log-day/index.jsx

```diff
@@ -34,25 +34,17 @@ const DAY_IN_MILLISECONDS = 1000 * 60 * 60 * 24;
  * classifier and the event itself (for rendering)
  *
  * @param {Array} logs sorted activity log items
- * @param {Number} activityId selected rewind operation
+ * @param {?Number} rewindId selected rewind operation
  * @returns {Array<String, ?Object>} pairs of [ classifier, event ]
  */
-const classifyEvents = ( logs, activityId ) => {
-	/** @type {Array<Array<Object, ?Object>>} contains pairs of [ log, next log ] **/
-	const logPairs = zip( logs, logs.slice( 1 ) );
-
-	return flatMap( logPairs, ( [ log, nextLog ] ) => {
-		if ( log.activityId === activityId ) {
-			return [ [ 'rewind-confirm-dialog', {} ], [ 'event', log ] ];
-		}
-
-		if ( nextLog && nextLog.activityId === activityId ) {
-			return [ [ 'event-before-dialog', log ] ];
-		}
-
-		return [ [ 'event', log ] ];
-	} );
-};
+const classifyEvents = ( logs, { rewindId = null } ) =>
+	// the zip pairs up each log item with the following log item in the stream or undefined if at end
+	flatMap( zip( logs, logs.slice( 1 ) ), ( [ log, nextLog = {} ] ) =>
```

Comment:
so you're...

- building up an array of pairs so you can easily match up an event with it's trailing event (the zip)
- using compact to build up an array where the first element is `[ 'rewind-confirm-dialog', {} ]` if the log's activity id matches the rewind id and `false` otherwise. The next element is either `[ 'timeline-break-event', log ]` if the next log entry matches the rewind id, and `[ 'event', log ]` otherwise. compact removes the possible leading false and then flatMap will flattens out the rest of the entries.

Reply:
Roger! Earlier I had the zip split out with another comment but I inlined it here. Actually this is probably more clear later on with more cases.

Do you think it would help to pull out the “timeline-break” into separate items instead of mashing them together here?